### PR TITLE
Load files function

### PR DESCRIPTION
**Pull request by Qing-Yang-1211**
Take in three input files instead of one now.

### CODE REVIEW

**Comment by ssantichaivekin** [comment]
Please merge from master.

File: ReconInput.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
+# ReconInput class
```
[the remark this answers is not in the document]
Comment (ssantichaivekin) [reply ReconInput.py:1]: It seems that Zach will be working on this. Let's wait for that change to go in.

**Comment by ssantichaivekin** [comment]
Please add some screenshots or more detailed description to the pull request.

**Comment by ssantichaivekin** [comment]
I seems odd to me that you are also uploading Zach's code. I think a better approach is for Zach to open a new pull request which add the `ReconInput` changes. Let's call the branch `recon-input-three-files`. Then, you could reset the history of your branch to his branch via `git reset` and open a new branch `gui-load-files`. After that, you can make a pull request to `recon-input-three-files`. If you have any comments or suggestions, you can make it to his pull request.

File: empress_gui.py [hunk 2]
```diff
@@ -53,59 +55,118 @@ def __init__(self, master):
...
+
+        # "Load Files" button 
+        # Loads in three input files (two .nwk and one .mapping)
+        # and displays the number of leaves in each tree and the entry boxes for setting DTL costs (next step)
+        self.file_to_load = tk.StringVar(self.input_frame) 
+        self.file_to_load.set("Load Files")
+        self.options = ["Load host tree file", "Load parasite tree file", "Load mapping file"]
+        self.load_file_list = tk.OptionMenu(self.input_frame, self.file_to_load, *self.options, command=self.load_input_files)
+        self.load_file_list.grid(row=0, column=0)
+        self.R = ReconInput.ReconInput()
```
[the remark this answers is not in the document]
Comment (Qing-Yang-1211) [reply empress_gui.py:76]: Done

File: empress_gui.py [hunk 3]
```diff
@@ -4,11 +4,13 @@
 from tkinter import messagebox
 from pathlib import Path
 import os
-import numpy as np
 import matplotlib.pyplot as plt
 from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg, NavigationToolbar2Tk
 from matplotlib.backend_bases import MouseEvent, key_press_handler
 import empress
+import ReconInput
```
[the remark this answers is not in the document]
Comment (Qing-Yang-1211) [reply empress_gui.py:11]: Yes. Ok!

File: ReconInput.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Qing-Yang-1211) [reply ReconInput.py:1]: Ok!

File: empress_gui.py [hunk 4]
```diff
@@ -53,59 +55,118 @@ def __init__(self, master):
...
+        Load in two .nwk files for the host tree and parasite tree, and one .mapping file. Display the number of tips for 
+        the trees and a message to indicate the successful reading of the tips mapping.
+        """ 
+        # Clicking on "Load host tree file" 
+        if self.file_to_load.get() == "Load host tree file":
+            # Allows loading a .newick file
+            self.host_file_path = None
+            # initialdir is set to be the current working directory
+            input_file = tk.filedialog.askopenfilename(initialdir=os.getcwd(), title="Select a host file")
+            if Path(input_file).suffix == '.nwk':
```
[the remark this answers is not in the document]
Comment (Qing-Yang-1211) [reply empress_gui.py:103]: I agree. Asked Zach to add that. :))

File: empress_gui.py [hunk 5]
```diff
@@ -53,59 +55,118 @@ def __init__(self, master):
...
+        # to display the numbers of tips for host and parasite trees
+        self.input_info_frame = tk.Frame(self.output_frame)
+        self.input_info_frame.grid(row=0, column=0, sticky="nsew")
+        self.input_info_frame.grid_columnconfigure(0, weight=1)
+        self.input_info_frame.grid_rowconfigure(0, weight=1)
+        self.input_info_frame.grid_rowconfigure(1, weight=1)
+        self.input_info_frame.grid_rowconfigure(2, weight=1)
+        self.input_info_frame.grid_propagate(False)
+
+        # "Load Files" button
```
[the remark this answers is not in the document]
Comment (Qing-Yang-1211) [reply empress_gui.py:68]: I have made the changes to grey out other unavailable options in the dropdown menu. :)) Will demonstrate to the group later. Please reply to me if there are other concerns or feedback!

File: empress_gui.py [hunk 6]
```diff
@@ -53,59 +55,118 @@ def __init__(self, master):
...
+        self.input_info_frame.grid_columnconfigure(0, weight=1)
+        self.input_info_frame.grid_rowconfigure(0, weight=1)
+        self.input_info_frame.grid_rowconfigure(1, weight=1)
+        self.input_info_frame.grid_rowconfigure(2, weight=1)
+        self.input_info_frame.grid_propagate(False)
+
+        # "Load Files" button 
+        # Loads in three input files (two .nwk and one .mapping)
+        # and displays the number of leaves in each tree and the entry boxes for setting DTL costs (next step)
+        self.file_to_load = tk.StringVar(self.input_frame)
```
[the remark this answers is not in the document]
Comment (Qing-Yang-1211) [reply empress_gui.py:71]: Done

[the remark this answers is not in the document]
Comment (Qing-Yang-1211) [reply empress_gui.py:71]: Changed to recon_input_file_option

File: ReconInput.py [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+
+    def __init__(self, host_tree = None, host_distances = None, parasite_tree = None, 
+                parasite_distances = None, phi = None):
+        self.host_tree = host_tree
+        self.host_distances = host_distances
+        self.parasite_tree = parasite_tree
+        self.parasite_distances = parasite_distances
+        self.phi = phi
+
+    def complete(self):
```
[the remark this answers is not in the document]
Comment (Qing-Yang-1211) [reply ReconInput.py:22]: Done

[the remark this answers is not in the document]
Comment (Qing-Yang-1211) [reply ReconInput.py:22]: It is a wrapper in Zach's code I think.

**Comment by Qing-Yang-1211** [comment]
> I seems odd to me that you are also uploading Zach's code. I think a better approach is for Zach to open a new pull request which add the `ReconInput` changes. Let's call the branch `recon-input-three-files`. Then, you could reset the history of your branch to his branch via `git reset` and open a new branch `gui-load-files`. After that, you can make a pull request to `recon-input-three-files`. If you have any comments or suggestions, you can make it to his pull request.

Actually, I realized that I have modified this "Load files" function quite significantly after adding the other wrappers for the reconciliation results to the gui. It is hard to separate the code for this "Load files" function from the others to reflect the meaningful changes in this branch. I am thus considering if it is possible to merge this pull request(and branch) to the pull request(and branch) for the reconciliation wrappers since the two things are very closely related. It won't be hard to review them together since the "Load files" function doesn't consist of a lot of code and it might be more efficient to review it together with the "reset" function(which is the main thing inside this load files function and is closely related to the reconciliation wrappers). What do you think? :))

**Comment by ssantichaivekin** [comment]
Although I generally think that separate functionalities should be separated into different branches, I agree it will be more convenient if you close this branch and only work on one branch. I also don't think your branch will be very large, so it should not be a problem for future code review. Still, it might be a good practice for you to work on multiple dependent branches at the same time.

**Pull request closed without merging**